airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/state/MemoryManager.kt

```diff
@@ -17,31 +19,49 @@ import kotlin.concurrent.withLock
  * TODO: Some degree of logging/monitoring around how accurate we're actually being?
```

Comment:
nit: add a `TODO: Be more fair. The current implementation will guarantee a first-come first-served order, which means if a function requests a large amount of memory that's not available yet, any subsequent call will block until the large memory request is satisfied, even though there may be enough available memory to serve the smaller requests`